VMEncryption/main/handle.py

```diff
@@ -579,6 +631,38 @@ def enable():
                 logger.log("No daemon found, trying to find the last non-query operation")
                 hutil.find_last_nonquery_operation = True
 
+        elif encryption_operation == CommonVariables.Migrate:
+            try:
+                logger.log("Migrate operation found. Starting migration flow.")
+                encryption_status = json.loads(disk_util.get_encryption_status())
+                if (encryption_status['data'] == 'NotEncrypted' or encryption_status['data'] == 'NotMounted') and encryption_status['os'] == 'NotEncrypted':
+                    logger.log("No device encrypted. Migration allowed")
+                    hutil.do_exit(exit_code=CommonVariables.success,
+                                  operation='Migrate',
+                                  status=CommonVariables.extension_success_status,
+                                  code=(CommonVariables.success),
+                                  message="Migration Allowed")
+                    migration_allowed, error = is_migration_allowed(disk_util, existing_passphrase_file)
+                    if not migration_allowed:
+                        hutil.do_exit(exit_code=CommonVariables.configuration_error,
+                                      operation='Migrate',
+                                      status=CommonVariables.extension_error_status,
+                                      code=(CommonVariables.configuration_error),
```

Comment:
Is 'Migrate' a valid operation type to exit with?  Expected this to be one of "Install/Enable/etc.." but if custom operations can be listed here that's OK.   Also maybe configuration_error could be changed to a new error code just for tracking any migrate failed telemetry specifically.

Reply:
good point. Though I have not observed any issue in my testing but it can cause the operation to not show up in guest agent telemetry. I will change the operation to Enable.
I can change to new error code, but we might have to add another error code in guest agent kusto query to not count it as extension failure. Will that be fine?